src/kernels/kernel.ts

```diff
@@ -661,7 +664,7 @@ abstract class BaseKernel<TKernelExecution extends BaseKernelExecution> implemen
     }
 }
 
-export class ThirdPartyKernel extends BaseKernel<ThirdPartyKernelExecution> {
+export class ThirdPartyKernel extends BaseKernel<ThirdPartyKernelExecution> implements IThirdPartyKernel {
```

Comment:
@rebornix both ThirdPartyKernel and internal Kernel are almost identical.
The internal kernel has 2 additoinal properties, that's the only difference now.